### PR TITLE
[TE]frontend - Add filtering support for Entity Monitoring tables

### PR DESCRIPTION
**Pull request by tejasajmera**
Filtering is supposed to work on column level with all possible data types in the table. It will do a substring match of the string passed in with each of the column entries; a return value of true would make table display the corresponding row.

Miscellaneous
1. Refactored `anomaliesDetails` to avoid doing the O(N) computation along with O(N) memory involved in converting object to array to be used in the template. Instead the template is modified to directly work with the object.
2. Refactored the columns in each of the table for entity monitoring to ensure reuse columns which can be shared.

Note to reviewers:
Lot of changes in `utils.js` are style related auto changed by the prettifier. Please focus on just 2 functions in that file -`isObject` and `checkForMatch`. Both are towards the end of the file.

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/utils/utils.js [hunk 1]
```diff
@@ -275,11 +277,68 @@ export function buildBounds(series, baseline, timeseries, useCurrent) {
...
+  }
+
+  if (typeof input === 'string') {
+    return input.includes(filterStr);
+  } else if (typeof input === 'number' || typeof input === 'boolean') {
+    return input.toString().includes(filterStr);
+  } else if (isObject(input)) {
+    for (const prop in input) {
+      if ({}.propertyIsEnumerable.call(input, prop)) {
+        if (checkForMatch(prop, filterStr) || checkForMatch(input[prop], filterStr)) {
```
Comment (zhangloo333) [inline thirdeye/thirdeye-frontend/app/utils/utils.js:322]: multiple level nest conditions will muddy up code's control flow. Also, it takes difficulty for later maintenance and result devs to spend more time on understanding state and logic. These two `if` statement are unnesstary, I suggest u can put them in one such as `if ({}.propertyIsEnumerable.call(input, prop) && (checkForMatch(prop, filterStr) || checkForMatch(input[prop], filterStr))`

**Review by zhangloo333 — CHANGES_REQUESTED** [review]
Others LGTM

File: thirdeye/thirdeye-frontend/app/utils/utils.js (hunk 1 above)
Comment (tejasajmera) [reply thirdeye/thirdeye-frontend/app/utils/utils.js:322]: I can club the `if's` but I don't necessarily agree it leads to improved maintenance or less time understanding the logic, as the logic is not really changed by clubbing the `if's`.

**Review by zhangloo333 — APPROVED** [review]

**Review by JoeFuentes-WebDev — APPROVED** [review]
Looks good to me

**Review by jihaozh — APPROVED** [review]

**Pull request merged**